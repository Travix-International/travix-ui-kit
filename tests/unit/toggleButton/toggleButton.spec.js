import { mount, shallow } from 'enzyme';
import React from 'react';
import ToggleButton from '../../../components/toggleButton/toggleButton';
import ToggleItem from '../../../components/toggleButton/toggleItem';

describe('ToggleButton', () => {
  const defaultMockedHandleSelect = jest.fn();

  it('renders null when no items or children are provided to ToggleButton', () => {
    const wrapper = shallow(<ToggleButton handleSelect={defaultMockedHandleSelect} />);
    expect(wrapper.type()).toEqual(null);
  });

  it('renders null when only one item is provided to ToggleButton', () => {
    const wrapper = shallow(
      <ToggleButton
        handleSelect={defaultMockedHandleSelect}
        items={['First item']}
      />
    );

    expect(wrapper.type()).toEqual(null);
  });

  it('renders null when only one children is provided to ToggleButton', () => {
    const wrapper = shallow(
      <ToggleButton handleSelect={defaultMockedHandleSelect}>
        <ToggleItem>First Item</ToggleItem>
      </ToggleButton>
    );

    expect(wrapper.type()).toEqual(null);
  });

  it('renders the ToggleButton component using items', () => {
    const items = [
      'First item',
      'Second item',
      'Third item',
    ];
    const wrapper = mount(<ToggleButton handleSelect={defaultMockedHandleSelect} items={items} />);

    const selectedItem = wrapper.find('.ui-toggle-button__item_active');

    expect(wrapper).toMatchSnapshot();
    expect(selectedItem.text()).toEqual(items[0]);
  });

  it('renders the ToggleButton component with the defaults', () => {
    const wrapper = mount(
      <ToggleButton handleSelect={defaultMockedHandleSelect}>
        <ToggleItem>First Item</ToggleItem>
        <ToggleItem>Second Item</ToggleItem>
        <ToggleItem>Third Item</ToggleItem>
      </ToggleButton>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the ToggleButton component with data-attributes', () => {
    const wrapper = mount(
      <ToggleButton data-gtm-id={123} handleSelect={defaultMockedHandleSelect}>
        <ToggleItem data-xivart-elm={'first-item'}>First Item</ToggleItem>
        <ToggleItem>Second Item</ToggleItem>
        <ToggleItem>Third Item</ToggleItem>
      </ToggleButton>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('passing a value in selectedIndex will select the proper index of the list of items', () => {
    const wrapper = mount(
      <ToggleButton handleSelect={defaultMockedHandleSelect} selectedIndex={2}>
        <ToggleItem>First Item</ToggleItem>
        <ToggleItem>Second Item</ToggleItem>
        <ToggleItem>Third Item</ToggleItem>
      </ToggleButton>
    );
    const selectedItem = wrapper.find('.ui-toggle-button__item_active');

    expect(wrapper).toMatchSnapshot();
    expect(selectedItem.text()).toEqual('Third Item');
  });

  it('when a "handleSelect" function is passed it gets executed when clicking on an option', () => {
    const mockedHandleSelect = jest.fn();
    const wrapper = mount(
      <ToggleButton handleSelect={mockedHandleSelect} selectedIndex={2}>
        <ToggleItem>First Item</ToggleItem>
        <ToggleItem>Second Item</ToggleItem>
        <ToggleItem>Third Item</ToggleItem>
      </ToggleButton>
    );

    const fakeEvent = { stopPropagation: jest.fn() };

    wrapper.find(ToggleItem).first().simulate('click', fakeEvent);

    expect(fakeEvent.stopPropagation).toHaveBeenCalledTimes(1);
    expect(mockedHandleSelect).toHaveBeenCalledTimes(1);
  });

  it('should render ToggleItem standalone', () => {
    const wrapper = mount(
      <ToggleItem>Self contained Toggle Item</ToggleItem>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return null if no handleClick function is passed to ToggleItem', () => {
    const wrapper = mount(
      <ToggleItem>
        Self contained Toggle Item
      </ToggleItem>
    );

    expect(wrapper.props().handleClick()).toBeNull();
  });

  it('when a "handleSelect" function is passed it gets executed when clicking on an option using items', () => {
    const mockedHandleSelect = jest.fn();
    const items = [
      'First item',
      'Second item',
      'Third item',
    ];
    const wrapper = mount(<ToggleButton handleSelect={mockedHandleSelect} items={items} />);

    const fakeEvent = { stopPropagation: jest.fn() };

    wrapper.find(ToggleItem).first().simulate('click', fakeEvent);

    expect(fakeEvent.stopPropagation).toHaveBeenCalledTimes(1);
    expect(mockedHandleSelect).toHaveBeenCalledTimes(1);
  });
});
