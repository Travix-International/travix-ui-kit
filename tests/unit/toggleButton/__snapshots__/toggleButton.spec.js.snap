// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton passing a value in selectedIndex will select the proper index of the list of items 1`] = `
<ToggleButton
  handleSelect={[MockFunction]}
  selectedIndex={2}
>
  <ul
    className="ui-toggle-button"
  >
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".0"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        First Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".1"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Second Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={true}
      handleClick={[Function]}
      key=".2"
    >
      <li
        className="ui-toggle-button__item ui-toggle-button__item_active"
        onClick={[Function]}
      >
        Third Item
      </li>
    </ToggleItem>
  </ul>
</ToggleButton>
`;

exports[`ToggleButton renders the ToggleButton component using items 1`] = `
<ToggleButton
  handleSelect={[MockFunction]}
  items={
    Array [
      "First item",
      "Second item",
      "Third item",
    ]
  }
  selectedIndex={0}
>
  <ul
    className="ui-toggle-button"
  >
    <ToggleItem
      active={true}
      handleClick={[Function]}
      key="0"
    >
      <li
        className="ui-toggle-button__item ui-toggle-button__item_active"
        onClick={[Function]}
      >
        First item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key="1"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Second item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key="2"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Third item
      </li>
    </ToggleItem>
  </ul>
</ToggleButton>
`;

exports[`ToggleButton renders the ToggleButton component with data-attributes 1`] = `
<ToggleButton
  data-gtm-id={123}
  handleSelect={[MockFunction]}
  selectedIndex={0}
>
  <ul
    className="ui-toggle-button"
  >
    <ToggleItem
      active={true}
      data-xivart-elm="first-item"
      handleClick={[Function]}
      key=".0"
    >
      <li
        className="ui-toggle-button__item ui-toggle-button__item_active"
        onClick={[Function]}
      >
        First Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".1"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Second Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".2"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Third Item
      </li>
    </ToggleItem>
  </ul>
</ToggleButton>
`;

exports[`ToggleButton renders the ToggleButton component with the defaults 1`] = `
<ToggleButton
  handleSelect={[MockFunction]}
  selectedIndex={0}
>
  <ul
    className="ui-toggle-button"
  >
    <ToggleItem
      active={true}
      handleClick={[Function]}
      key=".0"
    >
      <li
        className="ui-toggle-button__item ui-toggle-button__item_active"
        onClick={[Function]}
      >
        First Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".1"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Second Item
      </li>
    </ToggleItem>
    <ToggleItem
      active={false}
      handleClick={[Function]}
      key=".2"
    >
      <li
        className="ui-toggle-button__item"
        onClick={[Function]}
      >
        Third Item
      </li>
    </ToggleItem>
  </ul>
</ToggleButton>
`;

exports[`ToggleButton should render ToggleItem standalone 1`] = `
<ToggleItem
  active={false}
  handleClick={[Function]}
>
  <li
    className="ui-toggle-button__item"
    onClick={[Function]}
  >
    Self contained Toggle Item
  </li>
</ToggleItem>
`;
